mira/sources/vensim.py

```diff
@@ -0,0 +1,360 @@
+        old_var_name = var_declaration[0].strip()
+        text_expression = var_declaration[1]
+        new_var_expression_map[
+            old_name_new_pyname_map[old_var_name]
+        ] = text_expression
+
+    model_states = model_doc_df[model_doc_df["Type"] == "Stateful"]
+    concepts = {}
+    all_states = set()
+
+    # map between states and incoming/out-coming rate laws
+    state_rate_map = {}
+    # map between states and sympy version of the INTEG expression representing that state
+    state_sympy_map = {}
+
+    # process states and build mapping of state to input rate laws and output rate laws
+    for index, state in model_states.iterrows():
+        concept_state = state_to_concept(state)
+        concepts[concept_state.name] = concept_state
+        all_states.add(concept_state.name)
+        symbols[concept_state.name] = sympy.Symbol(concept_state.name)
+
+        state_name = state["Py Name"]
+        state_rate_map[state_name] = {"inputs": [], "outputs": []}
+        state_expr_text = new_var_expression_map[state_name]
+        state_arg_text = re.search("INTEG+ \( (.*),", state_expr_text).group(1)
+        state_arg_sympy = process_expression_text(state_arg_text, symbols)
+
+        # map of states to rate laws that affect the state
+        state_sympy_map[state_name] = state_arg_sympy
+
+        # Create a map of states and whether the rate-law/s involved with a state are going in
+        # or out of the state
+        if state_arg_sympy.args:
+            # if it's just the negation of a single symbol
+            if (
+                sympy.core.numbers.NegativeOne() in state_arg_sympy.args
+                and len(state_arg_sympy.args) == 2
+            ):
+                str_symbol = str(state_arg_sympy)
+                state_rate_map[state_name]["outputs"].append(str_symbol[1:])
+            else:
+                for rate_free_symbol in state_arg_sympy.args:
+                    str_rate_free_symbol = str(rate_free_symbol)
+                    if "-" in str_rate_free_symbol:
+                        # Add the symbol to outputs symbol without the negative sign
+                        state_rate_map[state_name]["outputs"].append(
+                            str_rate_free_symbol[1:]
+                        )
+                    else:
+                        state_rate_map[state_name]["inputs"].append(
+                            str_rate_free_symbol
+                        )
+        else:
+            # if it's just a single symbol (i.e. no negation), args property will be empty
+            state_rate_map[state_name]["inputs"].append(str(state_arg_sympy))
+
+    # process initials, just append 0 to each state to represent state at timestamp 0
+    mira_initials = {}
+    for state_name, state_concept in concepts.items():
+        initial = Initial(
+            concept=concepts[state_name].copy(deep=True),
+            expression=safe_parse_expr(state_name + "0"),
```

Comment:
If initial values are provided directly by the input model then instead of creating a new parameter, we can parse that value into a sympy Expr object and set it directly as the expression of the Initial.